### PR TITLE
Fix tool initialization detection

### PR DESCRIPTION
**Pull request by wtgodbe**
Similar to https://github.com/dotnet/coreclr/pull/11912, for https://github.com/dotnet/corefx/issues/25690

CC @weshaggard @mikedn @russellhadley

### CODE REVIEW

File: init-tools.cmd [hunk 1]
```diff
@@ -84,6 +85,9 @@ if not [%INIT_TOOLS_ERRORLEVEL%]==[0] (
 
 :: Create semaphore file
 echo Done initializing tools.
+if NOT exist "%BUILD_TOOLS_SEMAPHORE_DIR%" (
```
Comment (weshaggard) [inline init-tools.cmd]: Move this to the top as there there is another location that writes the semaphore file as well see line 34.

Comment (weshaggard) [reply init-tools.cmd]: I would also make int a singe line.

Comment (wtgodbe) [reply init-tools.cmd]: @weshaggard updated

**Comment by RussellHaley** [comment]
So I assumed I would need to git pull and then let it install once. However it never completed. 

https://pastebin.com/176CDFA6

It seems to start a download and then hangs. I left it there for 15 minutes and the file "C:\Users\russh\Git\dotnet\corefx\Tools\dotnetcli\dotnet-sdk-2.0.0-win-x64.zip" never grew larger than 1.5k. I checked performance monitor and there is no related network usage. A 63 MB FreeBSD bootonly iso downloaded in around 15 seconds. 

I killed the process and then tried again and now it is hung after another 20 mintues of waiting. I will kill this again and see if the aborted buildtools restore works correctly.

**Comment by RussellHaley** [comment]
I can confirm the change works correctly. The tools report as initialized in the second run. 
/*****************************************************/
C:\Users\russh\Git\dotnet\corefx>build.cmd
Tools are already initialized.
Running: C:\Users\russh\Git\dotnet\corefx\src\Native\build-native.cmd   x64  Debug  Windows_NT  --numproc 4  -portable  toolSetDir=c:\tools\clr
Tools are already initialized.
Running: C:\Users\russh\Git\dotnet\corefx\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false /l:BinClashLogger,Tools\net46\Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log /p:ConfigurationGroup=Debug /p:BuildPackages=false /p:GenerateNativeVersionInfo=true  /flp:v=normal  /flp2:warningsonly;logfile=msbuild.wrn  /flp3:errorsonly;logfile=msbuild.err  C:\Users\russh\Git\dotnet\corefx\src\Native\..\../build.proj /t:GenerateVersionHeader

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.07
Command execution succeeded.

**Pull request merged**